### PR TITLE
Split pandas pytests to prepare for GPU vs CPU metrics reporting

### PR DESCRIPTION
**Pull request by galipremsagar**
## Description

This PR splits the pandas tests job into 3 groups using `pytest-split`. The purpose of this PR is to reduce the runtime and avoid hangs being faced in running all the tests in one job along with profiling that will detect and extract metrics of GPU vs CPU execution in: https://github.com/rapidsai/cudf/pull/16754

As part of splitting the tests into separate groups we need `.test_durations` file which is 25 MB, hence I've added this file using git lfs.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -0,0 +1 @@
+python/cudf/cudf/pandas/scripts/.test_durations filter=lfs diff=lfs merge=lfs -text
```
Comment (bdice) [inline .gitattributes:1]: Is this change intentional? Why? I strongly oppose introducing LFS requirements to this repository.

**Review by bdice — CHANGES_REQUESTED** [review]
There's no PR description or linked issue so I can't tell what you aim to accomplish in this. We should find a solution that does not require git LFS.

**Comment by galipremsagar** [comment]
> There's no PR description or linked issue so I can't tell what you aim to accomplish in this. We should find a solution that does not require git LFS.

@bdice I just updated the PR title. The reason for using git LFS is the file size. I can move the file to s3 but this file will require future updates if the test suite changes drastically.

File: .gitattributes (hunk 1 above)
Comment (galipremsagar) [reply .gitattributes:1]: Responded here: https://github.com/rapidsai/cudf/pull/16743#issuecomment-2338166859

**Review by bdice — CHANGES_REQUESTED** [review]
Please remove `pytest-split` if we are using `pytest-shard` instead.

Please run all shards in one worker (serially) to avoid paying for the job setup cost on multiple GPU workers (scarce resources).

**Comment by galipremsagar** [comment]
@bdice This PR is not needed anymore as https://github.com/rapidsai/cudf/pull/16739 has all the necessary changes needed.
